### PR TITLE
[install] Bugfix 500 Internal Server Error 20.0-release

### PR DESCRIPTION
**Pull request by intralizee**
This pull request `fixes 500 Internal Server Error from happening after user submits admin mysql detail`.

The namespace references to the Database.class.inc in the php/installer directory was breaking the installer by some exceptions being thrown. Solution is to try catch the exceptions.

### CODE REVIEW

**Comment by kongtiaowang** [comment]
@intralizee 
I have tested it on Ubuntu and Centos, both get the same error "[Fri Jun 22 14:35:18.920845 2018] [php7:warn] [pid 23232] [client 132.216.237.16:49532] PHP Warning:  preg_match(): Compilation failed: nothing to repeat at offset 33 in /var/www/Loris/php/installer/Database.class.inc on line 76, referer: http://132.206.37.22/installdb.php
[Fri Jun 22 14:35:18.920916 2018] [php7:warn] [pid 23232] [client 132.216.237.16:49532] PHP Warning:  preg_match(): Compilation failed: nothing to repeat at offset 33 in /var/www/Loris/php/installer/Database.class.inc on line 76, referer: http://132.206.37.22/installdb.php
[Fri Jun 22 14:35:19.523620 2018] [php7:error] [pid 23232] [client 132.216.237.16:49532] PHP Fatal error:  Uncaught ConfigurationException: Config file config.xml does not exist in the defined locations. in /var/www/Loris/php/libraries/NDB_Config.class.inc:146\nStack trace:\n#0 /var/www/Loris/php/libraries/NDB_Config.class.inc(73): NDB_Config->configFilePath()\n#1 /var/www/Loris/php/libraries/NDB_Factory.class.inc(118): NDB_Config::singleton(NULL)\n#2 /var/www/Loris/php/libraries/NDB_Client.class.inc(54): NDB_Factory->config(NULL)\n#3 /var/www/Loris/htdocs/index.php(22): NDB_Client->initialize()\n#4 {main}\n  thrown in /var/www/Loris/php/libraries/NDB_Config.class.inc on line 146, referer: http://132.206.37.22/installdb.php"

**Comment by intralizee** [comment]
@kongtiaowang have you been installing Loris with the use existing database option checked?
I only tested the PR with the **Use existing database** and with the checkbox for "Check this if the database already exists but does not have Loris installed yet" selected. That case works for me

I wasn't supplied with MySQL and a root user when I got my VM as well. I'll try installing mysql locally on my development laptop to attempt your install configuration which I'm assuming is when a database does not exist yet.

**Comment by kongtiaowang** [comment]
@intralizee It usually provides to the new users to install new Loris, so without  "existing database" is necessary to test. I can give you an installing VM, It has everything you need.

File: htdocs/test.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (johnsaigle) [inline htdocs/test.php]: What's this file here? Why do we need this information in the codebase?

Comment (intralizee) [reply htdocs/test.php]: oops that shouldn't be there.

**Comment by intralizee** [comment]
@kongtiaowang the installer script should now work. 
I wasn't able test on the VM you supplied but I did test on my local machine with mysql and a root user.

File: php/installer/Installer.class.inc [hunk 2]
```diff
@@ -185,21 +186,30 @@ class Installer
...
-        ) {
+        try {
+            if ($DB->connectAndCreate(
+                $values['dbname'],
+                $values['dbadminuser'],
+                $values['dbadminpassword'],
+                $values['dbhost']
+            ) == false
+            ) {
+                if (!empty($DB->lastErr)) {
```
Comment (johnsaigle) [inline php/installer/Installer.class.inc]: This looks like it does the same thing as the code below. Isn't the try...catch the only part that's needed here?

File: php/installer/Installer.class.inc [hunk 3]
```diff
@@ -214,21 +224,32 @@ class Installer
...
-        ) {
+
+        try {
+            if ($DB->connect(
+                $values['dbname'],
+                $values['dbadminuser'],
+                $values['dbadminpassword'],
+                $values['dbhost']
+            ) == false
+            ) {
```
Comment (johnsaigle) [inline php/installer/Installer.class.inc]: Same here. The code in the `try` block seems redundant.

File: php/installer/Installer.class.inc (hunk 2 above)
Comment (intralizee) [reply php/installer/Installer.class.inc]: You're right, will remove.

**Comment by driusan** [comment]
@intralizee Looks fine to me, but has the latest iteration been tested by someone else?

**Comment by intralizee** [comment]
@driusan I tested it on my local machine when creating a non-existing db. Also tested when using an already existing db for installation. I don't believe it's been tested by someone else. There was a conflict when using with Shen's VM that I never figured out and am unable to reproduce.

**Comment by kongtiaowang** [comment]
@intralizee @driusan I am testing this, it works fine on Ubuntu, I will test it on Centos later.
If everything is good, I will add the passmanualtest.

**Comment by intralizee** [comment]
Added a fix where an invalid password resulted in a 500 error. @um4r12 helped.

**Pull request merged**